openforcefield/utils/toolkits.py

```diff
@@ -2948,6 +2949,161 @@ def compute_partial_charges_am1bcc(self, molecule):
+        """
+        Process a SQM output file containing bond orders, and return a dict of the form
+        dict[atom_1_index, atom_2_index] = fractional_bond_order
+
+        Parameters
+        ----------
+        file_path : str
+            File path for sqm output file
+
+        Returns
+        -------
+        bond_orders : dict[(int, int)]: float
+            A dictionary where the keys are tuples of two atom indices and the values are
+            floating-point bond orders. The keys are sorted in ascending order, such that
+            the lower atom index is key[0] and the higher is key[1].
+        """
+
+        # Example sqm.out section with WBOs:
+        #  Bond Orders
+        #
+        #   QMMM:    NUM1 ELEM1 NUM2 ELEM2      BOND_ORDER
+        #   QMMM:       2   C      1   C        1.41107532
+        #   QMMM:       3   C      1   C        1.41047804
+        # ...
+        #   QMMM:      15   H     13   H        0.00000954
+        #   QMMM:      15   H     14   H        0.00000813
+        #
+        #            --------- Calculation Completed ----------
+
+        data = open(file_path).read()
+
+        begin_sep = ''' Bond Orders
+ 
+  QMMM:    NUM1 ELEM1 NUM2 ELEM2      BOND_ORDER
+'''
+        end_sep = '''
+
+           --------- Calculation Completed ----------
+'''
+        fbo_lines = data.split(begin_sep)[1].split(end_sep)[0].split('\n')
+        bond_orders = dict()
+        for line in fbo_lines:
+            linesp = line.split()
+            atom_index_1 = int(linesp[1])
+            atom_index_2 = int(linesp[3])
+            bond_order = float(linesp[5])
+            # To make lookup easier, we identify bonds as integer tuples with the lowest atom index
+            # first and the highest second.
+            index_tuple = tuple(sorted([atom_index_1, atom_index_2]))
+            bond_orders[index_tuple] = bond_order
+        return bond_orders
+
+    def assign_fractional_bond_orders(self, molecule, bond_order_model=None):
+        """
+        Update and store list of bond orders this molecule. Bond orders are stored on each
+        bond, in the `bond.fractional_bond_order` attribute.
+
+        .. warning :: This API is experimental and subject to change.
+
+        Parameters
+        ----------
+        molecule : openforcefield.topology.molecule Molecule
+            The molecule to assign wiberg bond orders to
+        bond_order_model : str, optional, default=None
+            The charge model to use. Only allowed value is 'am1-wiberg'. If None, 'am1-wiberg' will be used.
+
+         """
+
+        # Find the path to antechamber
+        # TODO: How should we implement find_executable?
+        ANTECHAMBER_PATH = find_executable("antechamber")
+        if ANTECHAMBER_PATH is None:
+            raise (IOError("Antechamber not found, cannot run "
+                           "AmberToolsToolkitWrapper.assign_fractional_bond_orders()"))
+
+        if len(molecule._conformers) == 0:
```

Comment:
I understand this point but this could be a pain for users, is it not better to assume we need to generate a conformer if one is not supplied and just do this for them?

Reply:
Good call, and actually this parallels an API change that we'll make with `compute_partial_charges`. I'll make this change.

Reply:
Addresses in 8bd7fec